fibers.js

```diff
@@ -0,0 +1,25 @@
+const Future = require('fibers/future');
+const fs = Future.wrap(require('fs'));
+const prependDir = (file) => require('path').join(process.argv[2], file)
+
+const main = () => {
+  Future.task(function() {
+    try {
+      process.stdout.write(
+        fs
+        .readFileFuture(prependDir('index.txt'))
+        .wait()
+        .toString()
+        .replace(/\n$/,'')
+        .split('\n')
+        .map((x) => fs.readFileFuture(prependDir(x)).wait().toString())
+        .join(''))
```

Comment:
Besides I think Rx.js can be used to for this task.
I don't like Rx at all, but it would be great addition